tests/test_sim.py

```diff
@@ -516,6 +516,45 @@ def process():
                 self.assertEqual((yield self.o), 0)
             sim.add_sync_process(process)
 
+    def setUp_clock_phase(self):
+        self.m = Module()
+        self.m.domains.phase0 = ClockDomain()
+        self.m.domains.phase90 = ClockDomain()
+        self.m.domains.phase180 = ClockDomain()
+        self.m.domains.phase270 = ClockDomain()
+        self.m.domains.check = ClockDomain()
+
+        self.expected = [
+            [0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0],
+            [0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1],
+            [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0],
+        ]
+
+    def test_clock_phase(self):
+        self.setUp_clock_phase()
+        with self.assertSimulation(self.m) as sim:
+            period=1
+            sim.add_clock(period/8, phase=0,          domain="check")
+            sim.add_clock(period,   phase=0*period/4, domain="phase0")
+            sim.add_clock(period,   phase=1*period/4, domain="phase90")
+            sim.add_clock(period,   phase=2*period/4, domain="phase180")
+            sim.add_clock(period,   phase=3*period/4, domain="phase270")
+
+            def proc():
+                clocks = [
+                    self.m._domains["phase0"].clk,
+                    self.m._domains["phase90"].clk,
+                    self.m._domains["phase180"].clk,
+                    self.m._domains["phase270"].clk
```

Comment:
`_domains` is an internal member and shouldn't be accessed (above comment explains my original intent)

Reply:
I found inspiration for this in [test_hdl_dsl.py:778](https://github.com/amaranth-lang/amaranth/blob/c83b51db6daf3b73fb2406fdeecf6ef7486bb0be/tests/test_hdl_dsl.py#L778), but that tests the internals.